setup/k8s/djangodb.yaml

```diff
@@ -0,0 +1,202 @@
+# CM-8 & CM-9: Defines the application's components within the AKS cluster
+# SC-13: Ensure secure key management practices
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres-django-db
+  namespace: otto  
+spec:
+  serviceName: "postgres-django-service"
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgres-django-db
+  template:
+    metadata:
+      labels:
+        app: postgres-django-db
+        tier: backend    
+    spec:
+      containers:
+        - name: postgres-django-db
+          image: postgres:16
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: azure-keyvault-secrets
+                  key: djangodbpasswordkey
+            - name: POSTGRES_DB
+              valueFrom:
+                configMapKeyRef:
+                  name: otto-configmap
+                  key: DJANGODB_NAME
+            - name: POSTGRES_USER
+              valueFrom:
+                configMapKeyRef:
+                  name: otto-configmap
+                  key: DJANGODB_USER
+            - name: POSTGRES_MAX_CONNECTIONS
+              value: "500"
+          ports:
+            - containerPort: 5432
+          resources:
+            requests:
+              cpu: "500m"
+              memory: "4Gi"
+            limits:
+              cpu: "500m"
```

Comment:
Should we consider bumping up the CPU and memory limits before scaling horizontally?

Reply:
Add parameters. There is some give but for databases, the limits should not be significantly different than requests.